agent/exec/dockerapi/docker_client_stub.go

```diff
@@ -0,0 +1,51 @@
+package dockerapi
+
+import (
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/events"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/client"
+	"golang.org/x/net/context"
+	"io"
+	"time"
+)
+
+// StubAPIClient implements the client.APIClient interface, but allows
+// you to specify the behavior of each of the methods.
+type StubAPIClient struct {
+	client.APIClient
+	_ContainerCreate  func(_ context.Context, config *container.Config, hostConfig *container.HostConfig, networking *network.NetworkingConfig, containerName string) (container.ContainerCreateCreatedBody, error)
```

Comment:
Rather than use the `_` prefix, like you would in python, just call these `ContainerCreateFn`, to signify they are a function variable.